ARE/middleware/src/main/java/eu/asterics/mw/are/AsapiSupport.java

```diff
@@ -1791,55 +1793,86 @@ public boolean deleteModelFile(String filename) throws AREAsapiException {
     }
 
     /**
-     * Stores the XML model specified by the string parameter in the file
-     * specified by the filename parameter
-     * 
-     * @param modelInXML
-     *            the XML model as a String
+     * Stores data specified by the string parameter in the file specified by
+     * the path/filename parameter
+     *
+     * @param data
+     *            the data to be stored
+     * @param path
+     *            the path where the file should be stored
      * @param filename
-     *            the name of the file the model is to be stored
+     *            the name of the file to be stored
+     * @param charset
+     *            the charset the file should be stored with
      * @throws AREAsapiException
      *             if the file cannot be created or if the model cannot be
      *             stored
      */
-    public void storeModel(String modelInXML, String filename) throws AREAsapiException {
-        // First check if the model is a valid XML model
+    public void storeData(String data, String path, String filename, Charset charset) throws AREAsapiException {
 
-        File fileName = new File(ResourceRegistry.MODELS_FOLDER + "/" + filename);
-        File modelsDir = new File(ResourceRegistry.MODELS_FOLDER);
-        if (!fileName.exists()) {
+        File file = new File(path + "/" + filename);
+        File directory = new File(path);
```

Comment:
regarding path: you mean this line I should change, correct?
So with
`File filenameAsFile=ResourceRegistry.toFile(ResourceRegistry.getInstance().getResource("webservice/saved/data.json", RES_TYPE.DATA));` I would create a file in the folder `data/webservice/saved` folder, correct? Here we should maybe think about where to save data from the different webapps? in the webapps-folder of the webapp which want to save somethin or e.g. directly in `data/saved`?

Reply:
I thought the function would simply store a file in the ARE/data folder.
So it would simply be the same as https://github.com/asterics/AsTeRICS/blob/master/ARE/middleware/src/main/java/eu/asterics/mw/are/AsapiSupport.java#L1529

and just exchange the RES_TYPE to RES_TYPE.DATA

Having this we could edit, change config files of certain plugins directly via a webapplication.
But you are right, we should think about a good location for webapps specific data.